src/Devtools/Brittany.hs

```diff
@@ -0,0 +1,65 @@
+module Devtools.Brittany
+  ( Brittany.Config
+  , testTree
+  )
+where
+
+import           Control.Monad.Trans.Maybe      ( runMaybeT )
+import           Data.Functor.Identity          ( Identity )
+import           Devtools.Prelude
+import           Language.Haskell.Brittany      ( BrittanyError(..) )
+import           Test.Tasty.MGolden             ( goldenTest )
+
+import qualified Data.Text.IO                  as Text
+import qualified Language.Haskell.Brittany     as Brittany
+import qualified System.Path                   as Path
+import qualified System.Path.Directory         as Path
+import qualified Test.Tasty                    as Tasty
+
+testTree :: Maybe Path.RelFile -> [Path.RelFile] -> IO Tasty.TestTree
+testTree userConfigFile files = do
+  selectedConfig <- maybe (pure defaultConfig) getUserConfig userConfigFile
+
+  pure $ mkTestTree selectedConfig files
+
+mkTestTree :: Brittany.Config -> [Path.RelFile] -> Tasty.TestTree
+mkTestTree config = Tasty.testGroup "Brittany" . fmap singleTest
+ where
+  singleTest :: Path.RelFile -> Tasty.TestTree
+  singleTest (Path.toString -> filePath) = goldenTest filePath filePath $ do
+    content <- Text.readFile filePath
+
+    either (fail . show . fmap showBrittanyError) pure
+      =<< Brittany.parsePrintModule config content
+
+showBrittanyError :: BrittanyError -> String
+showBrittanyError = \case
+  ErrorInput         str -> str
+  ErrorUnusedComment str -> str
+  LayoutWarning      str -> str
+  ErrorUnknownNode str _ -> str
+  ErrorMacroConfig str _ -> "When parsing inline config: " <> str
+  ErrorOutputCheck       -> "Output is not syntactically valid."
+
+defaultConfig :: Brittany.Config
+defaultConfig = Brittany.staticDefaultConfig
```

Comment:
Just FYI @mbj 

The data-files way of bundling a YAML configuration with the project that I had before was based on the guide below. The downside to it is the following:

-  The data file had to be in a directory within the project such as `data` while using `stack` as a build tool. Attempting having them at the root would lead to confusing errors

- The data files direction works well when the project is going to be used as an executable. If you attempt to use it as a library import; the data files wouldn't be accessible. You would need to do some more work to retrieve them.


http://neilmitchell.blogspot.com/2008/02/adding-data-files-using-cabal.html